Phonebook.Frontend/src/app/modules/table/components/table/table.component.ts

```diff
@@ -110,11 +110,15 @@ export class TableComponent implements OnInit, OnDestroy {
   }
 
   public refreshTable() {
+    const whichColumns = TableState.visibleBigColumns;
+    if (this.isSizeSmall === true) {
+      return whichColumns === TableState.visibleSmallColumns;
+    }
     this.dataSource
       .refresh(
         this.store.selectSnapshot(SearchState.searchTerm).trim(),
         this.store.selectSnapshot(SearchState.searchFilters),
-        this.store.selectSnapshot(TableState.visibleBigColumns),
+        this.store.selectSnapshot(whichColumns),
```

Comment:
@DanielHabenicht I don't know whether this is the most elegant way of doing this. Simply adding a fourth selectSnapshot wouldn't be the best either since one would have to add a fourth argument to refresh() and that leads to adding a lot more arguments and properties in other files as well.

Reply:
Can you just use `this.displayedColumns`?

Reply:
Nevermind, this is shorter:
```suggestion
        this.store.selectSnapshot(this.isSizeSmall ? TableState.visibleSmallColumns : TableState.visibleBigColumns),
```

Reply:
Normally I would say that we should refactor this into the state but we dont have the time for it.

Reply:
You can remove the whichColun variable